kse/src/org/kse/gui/actions/VerifyCertificateAction.java

```diff
@@ -126,13 +152,125 @@ private void verifyChain(KeyStoreHistory keyStoreHistory, String alias)
+			throws OperatorCreationException, OCSPException, MalformedURLException, IOException, HeadlessException,
+			CertPathValidatorException, KeyStoreException, NoSuchAlgorithmException, CertificateException {
+
+		X509Certificate issuer = null;
+		KeyStore trustStore = getKeyStore(keyStoreHistory);
+		Enumeration<String> enumeration = trustStore.aliases();
+		while (enumeration.hasMoreElements()) {
+			String tempAlias = enumeration.nextElement();
+			X509Certificate cert = (X509Certificate) trustStore.getCertificate(tempAlias);
+			try {
+				certificateEval.verify(cert.getPublicKey());
+				issuer = cert;
+				break;
+			} catch (InvalidKeyException | CertificateException | NoSuchAlgorithmException | NoSuchProviderException
+					| SignatureException e) {
+				// ignore
+			}
+		}
+
+		if (issuer == null) {
+			throw new CertPathValidatorException(res.getString("VerifyCertificateAction.trustStoreEmpty.message"));
+		}
+		OCSPReq request = makeOcspRequest(issuer, certificateEval);
+		OCSPResp response = requestOCSPResponse(ocspUrl, request);
+		if (isGoodCertificate(response, issuer, certificateEval)) {
 			JOptionPane.showMessageDialog(frame, res.getString("VerifyCertificateAction.OcspSuccessful.message"),
 					res.getString("VerifyCertificateAction.Verify.Title") + " " + alias,
 					JOptionPane.INFORMATION_MESSAGE);
 		}
 	}
 
+	public static OCSPReq makeOcspRequest(X509Certificate caCert, X509Certificate certToCheck)
+			throws OCSPException, OperatorCreationException, CertificateEncodingException {
+		DigestCalculatorProvider digCalcProv = new JcaDigestCalculatorProviderBuilder().setProvider(BOUNCY_CASTLE.jce())
+				.build();
+
+		CertificateID certId = new JcaCertificateID(digCalcProv.get(CertificateID.HASH_SHA1), caCert,
+				certToCheck.getSerialNumber());
+
+		OCSPReqBuilder gen = new OCSPReqBuilder();
+		gen.addRequest(certId);
+		return gen.build();
+	}
+
+	public OCSPResp requestOCSPResponse(String url, OCSPReq ocspReq) throws IOException, MalformedURLException {
+		byte[] ocspReqData = ocspReq.getEncoded();
+
+		HttpURLConnection con = (HttpURLConnection) new URL(url).openConnection();
+		try {
+			con.setRequestProperty("Content-Type", "application/ocsp-request");
+			con.setRequestProperty("Accept", "application/ocsp-response");
+			con.setDoInput(true);
+			con.setDoOutput(true);
+			con.setUseCaches(false);
+			try (OutputStream out = con.getOutputStream()) {
+				IOUtils.write(ocspReqData, out);
+				out.flush();
+			}
+			byte[] responseBytes = IOUtils.toByteArray(con.getInputStream());
+			OCSPResp ocspResp = new OCSPResp(responseBytes);
+			return ocspResp;
+		} finally {
+			if (con != null) {
+				con.disconnect();
+			}
+		}
+	}
+
+	public boolean isGoodCertificate(OCSPResp ocspResp, X509Certificate caCert, X509Certificate eeCert)
+			throws OperatorCreationException, OCSPException, CertificateEncodingException, CertPathValidatorException {
```

Comment:
OperatorCreationException and CertificateEncodingException can be removed

Reply:
Fixed.